### ISSUE TITLE
Release version 1.0.0 (i.e., publish gem)

### ISSUE THREAD

**Issue by rmoriz** [middleman/middleman-asciidoc#4]
currently `middleman-asciidoc` is not available via rubygems.org.

**Comment by mojavelinux** [comment]
This is because we were waiting for a middleman 4 release. I see that there is now an alpha. If we can verify this gem works with middleman 4 alpha, then I think we're ready to put out our own alpha as well. Can you confirm?

**Comment by mojavelinux** [comment]
Middleman 4 is now in beta release, which means it's time to start thinking about releasing this gem. We'll need to update the versions in the gemspec (particularly moving to Asciidoctor 1.5.2) and verify it all still works. Then we can move forward with a gem release.

(Related PR https://github.com/middleman/middleman-asciidoc/pull/5)

**Comment by mojavelinux** [comment]
Middleman v4 is now out, so we'll proceed with this release.

**Comment by drallgood** [comment]
Anything we should address before we release this gem?

**Comment by mojavelinux** [comment]
@drallgood Great question. I did a full code review and found a few issues, which I scheduled for 1.0.0 and submitted proposed fixes.

Mostly it was getting the extension back in line with how it worked in Middleman 3. I also wanted to get the page-layout setting right. As you can see from the pull request, the extension should respond to the following settings:

* `:page-layout: default` - use the page layout named "default"
* `:page-layout!:` - generate a standalone document (don't use a layout, use the Asciidoctor default exterior)
* `:page-layout:` - use the auto-detected layout
* *(not specified)* - use the auto-detected layout

I also made sure the certain built-in attributes, such as title, author, etc populated the page metadata so that you can define a page without front matter (this wasn't quite working correctly).

Finally, it is possible to set arbitrary page metadata using the `page-` prefix. That means you can define tags for the blog using:

```
= Blog Title
Author Name
:page-tags: announcement,release
```

I'm sure there's a lot more things we can still do, but once those pull requests are merged, we should be off to a solid start.

**Comment by mojavelinux** [comment]
We may need to think about whether the base_dir setting is right. I am seeing that Middleman is now invoking Tilt in such a way that the docdir attribute gets set, so that seems to put us in a better place, but I think we need a feature to test specifically for nested includes.

**Comment by drallgood** [comment]
Great! I'm going to check those pull requests out and report back.

When I'm back in the office on Monday, I'll also give this another try with the changes we've made. The new documentation we're writing isn't that huge yet, but it should give us an idea whether things like nested images and cross references are working as they should.

**Comment by mojavelinux** [comment]
Fantastic!

**Comment by mojavelinux** [comment]
@drallgood I had forgotten one of the pull requests, which I sent this morning, just to give you a heads up. Let's aim to to a first release at the end of the week if all goes well.

**Comment by marcobehler** [comment]
Hi @mojavelinux ,
any news on this one?

**Comment by mojavelinux** [comment]
@marcobehler My stack of tasks got a bit piled up and I'm working on playing catch up. This task is on that list. I need to address #24, then follow up on note I jotted down about testing nested includes before we can release.

**Comment by mojavelinux** [comment]
I won't set #18 as a blocker, but I would like to see it completed soon after the release.

**Comment by everflux** [comment]
As #24 and #25 are done, a 1.0.0 release would really be great!

**Comment by mojavelinux** [comment]
There's another issue that needs to be resolved before we are ready for 1.0.0. See #27. A fix and tests on the way.

**Comment by mojavelinux** [comment]
Should we start with version 4.0.0 to make it clear this plugin is for Middleman 4 (and was split off from Middleman after Middleman 3) or should we stick with 1.0.0?

**Comment by mojavelinux** [comment]
One the fix for #26 is merged, I think we'll be good to go. Please test the latest changes (even better if you include the fix for #26, which is PR #29).

It would really be awesome if we had at least some of the docs done for the 1.0.0 release (#18). If nothing else, the following should be documented:

* how to set page layout and other metadata from AsciiDoc header (see cucumber tests for example)
* document that imagesdir matches `:images_dir` option by default, but can be changed using attributes
* document that base_dir defaults to docdir, but can be overridden using `:base_dir` option (e.g., `activate :asciidoc, base_dir: app.source_dir.expand_path`)

**Comment by mojavelinux** [comment]
I've published a release candidate. See https://rubygems.org/gems/middleman-asciidoc.

I'd like to get some feedback that it's in good working order and fill in a few of the docs before 1.0.0. But that should at least give you a start.

**Comment by mojavelinux** [comment]
I've documented all the aforementioned behavior in https://github.com/middleman/middleman-asciidoc/pull/36. Please review.

**Comment by mojavelinux** [comment]
1.0.0.rc.2 is now available.

**Comment by mojavelinux** [comment]
Now that v2.0.0 of the jekyll-asciidoc plugin is available, I'd like to align the behavior of this plugin with that one so we can move forward together. Then, I'll move forward with releasing 1.0.0.

**Comment by mojavelinux** [comment]
1.0.0.rc.3 is now available. I'm close to having everything wrapped up and aligned with the Jekyll plugin.

**Comment by mojavelinux** [comment]
I just released 1.0.0.rc.8. This is a last call for anyone interesting in testing this extension before the 1.0.0 release goes out. If I don't hear any objections, I'll make the release shortly after the start of the year.

**Comment by mojavelinux** [comment]
I added a few tests for the integration with Asciidoctor Diagram, which I wanted to get done before the 1.0.0 release. I may also add a few more tests for the integration with Middleman Blog. Otherwise, we're all ready for the 1.0.0 release!

**Comment by mojavelinux** [comment]
I'm happy to announce that 1.0.0 has been released!

https://rubygems.org/gems/middleman-asciidoc/versions/1.0.0
